source/extensions/dynamic_modules/sdk/rust/src/lib.rs

```diff
@@ -3227,6 +3227,57 @@ pub trait EnvoyNetworkFilter {
 
   /// Disable or enable connection close handling for this filter.
   fn disable_close(&mut self, disabled: bool);
+
+  /// Close the connection with termination details.
+  fn close_with_details(
+    &mut self,
+    close_type: abi::envoy_dynamic_module_type_network_connection_close_type,
+    details: &str,
+  );
+
+  /// Get the requested server name (SNI).
+  /// Returns None if SNI is not available.
+  fn get_requested_server_name(&self) -> Option<String>;
+
+  /// Get the direct remote (client) address and port without considering proxy protocol.
+  /// Returns None if the address is not available or not an IP address.
+  fn get_direct_remote_address(&self) -> Option<(String, u32)>;
+
+  /// Get the SSL URI SANs from the peer certificate.
+  /// Returns an empty vector if the connection is not SSL or no URI SANs are present.
+  fn get_ssl_uri_sans(&self) -> Vec<String>;
```

Comment:
this comes with an allocation each time it being called, so maybe we might want to think about how to avoid it